app.js

```diff
@@ -31,191 +35,235 @@ app.use(express.json({ verify: VerifyDiscordRequest(process.env.PUBLIC_KEY) }));
+                        objectName,
+                    });
+
+                    // Remove game from storage
+                    delete activeGames[gameId];
+                    // Update message with token in request body
+                    const endpoint = `webhooks/${process.env.APP_ID}/${req.body.token}/messages/${req.body.message.id}`;
+
+                    try {
+                        // Send results
+                        await res.send({
+                            type: InteractionResponseType.CHANNEL_MESSAGE_WITH_SOURCE,
+                            data: { content: resultStr },
+                        });
+                        // Update ephemeral message
+                        await DiscordRequest(endpoint, {
+                            method: "PATCH",
+                            body: {
+                                content: `Nice choice ${getRandomEmoji()}`,
+                                components: [],
+                            },
+                        });
+                    } catch (err) {
+                        console.error("Error sending message:", err);
+                    }
+                }
             }
         }
-    }
-    return res.status(404).send({ message: "Unknown action" });
-});
+        return res.status(404).send({ message: "Unknown action" });
+    });
 
-app.listen(PORT, () => {
-    console.log("Listening on port", PORT);
+    app.listen(PORT, () => {
+        console.log("Listening on port", PORT);
 
-    // Check if guild commands from commands.js are installed (if not, install them)
-    HasGuildCommands(config.APP_ID, config.GUILD_ID, [
-        TEST_COMMAND,
-        CHALLENGE_COMMAND,
-    ]);
+        // Check if guild commands from commands.js are installed (if not, install them)
+        HasGuildCommands(config.APP_ID, config.GUILD_ID, [
+            TEST_COMMAND,
+            CHALLENGE_COMMAND,
+        ]);
+    });
+};
+
+// CLI definition
+const y = yargs(hideBin(process.argv));
+y.scriptName("elprimobot");
+y.option("verbose", {
+    alias: "v",
+    type: "boolean",
+    description: "Run with verbose logging",
+});
+y.command({
+    command: "leetcode-daily",
+    description: "Sends leetcode daily problem to a Discord channel",
+    handler: async () => {
+        await postDailyMessages();
+    },
+});
+y.command({
+    command: "stats-daily",
+    description: "Sends Discord users daily chat activity stats",
+    handler: async () => {
+        const stats = new Stats(discordClient);
+        await stats.postDailyStats();
+    },
 });
+y.command({
+    command: "stats-weekly",
+    description: "Sends Discord users weekly chat activity stats",
+    handler: async () => {
+        const stats = new Stats(discordClient);
+        await stats.postWeeklyStats();
+    },
+});
+y.command({
+    command: "daemon",
+    description: "Runs the all the features in blocking mode",
+    handler: async () => {
+        await runDaemon();
+    },
+});
```

Comment:
is it possible to move these to their own file?

Reply:
I think the code as it's now is a bit messy. Do you think we can do it after your merge?

Reply:
sure thing

Reply:
Issue reported #60